tests/examples/test_all.py

```diff
@@ -92,10 +106,19 @@ def test_example_2_base_time_add_runs(example_2_base_time_add_path: Path) -> Non
     assert_run_in_parallel([controller_path, solver_path])
 
 
-def test_example_example_3_tsolver(
-    example_3_tsolver: Path, install_matplotlib: None
-) -> None:
-    controller_path = example_3_tsolver / "Tsolver.py"
-    solver_path = example_3_tsolver / "Controller.py"
+def test_example_example_3_tsolver(example_3_tsolver: Path) -> None:
+    install_dependencies(example_3_tsolver)
 
-    assert_run_in_parallel([controller_path, solver_path])
+    t_solver_path = example_3_tsolver / "Tsolver.py"
+    controller_path = example_3_tsolver / "Controller.py"
+
+    assert_run_in_parallel([controller_path, t_solver_path])
+
+
+def test_4_coupled_solver(example_4_coupled_solver: Path) -> None:
+    install_dependencies(example_4_coupled_solver)
+
+    coupler_script_path = example_4_coupled_solver / "coupler_script.py"
+    tsolver_script_path = example_4_coupled_solver / "tsolver_script.py"
+
+    assert_run_in_parallel([coupler_script_path, tsolver_script_path])
```

Comment:
and then
`assert_run_in_parallel([coupler_script_path, tsolver_script_path, emsolver_script_path])`